libs/@blockprotocol/type-system/rust/src/schema/data_type/constraint/mod.rs

```diff
@@ -313,4 +316,52 @@ mod tests {
 
         assert_eq!(errors, actual_errors);
     }
+
+    pub(crate) fn intersect_schemas(
+        schemas: impl IntoIterator<Item = JsonValue>,
+    ) -> Result<Vec<JsonValue>, Report<ResolveClosedDataTypeError>> {
+        let schemas = schemas
+            .into_iter()
+            .map(|schema| serde_json::from_value(schema).expect("Failed to parse schema"))
```

Comment:
I see copilot wrote these expect messages :P

Reply:
What's wrong with this message?

Reply:
shouldn't it be: "should be able to parse schema"?

> We recommend that expect messages are used to describe the reason you expect the Result should be Ok.

from: https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.expect

You expect that the schema is correct, not that it is incorrect.

Reply:
Personally I find this quite readable (ignoring the `untagged` thing 🙄):
> `Failed to parse schema: Error("data did not match any variant of untagged enum ValueConstraints", line: 0, column: 0)`

We use this format pretty scattered in the whole code base. As long as it's not `expect("schema to be valid")` I think we're good. I'm up for consistently change expect messages in the code base by actively searching for `.expect` (or even better use `Result` in tests instead, although this could become verbose). The other related tests in this crate uses the same format so I think we should just leave it untouched in this PR.

Reply:
Actually, I created H-3498 for that: Consistently use expect message in code base